### PR TITLE
Refine self_attention_fuse pass with removing the size restriction

### PR DESCRIPTION
**Pull request by abenmao**
### PR types
Bug fixes

### PR changes
Others

### Description
remove the size restriction in self_attention_fuse pass. Fix PaddleOCR v4 on AVX512 CPU.

### CODE REVIEW

**Review by YangQun1 — DISMISSED** [review]

**Review by xinyu-intel — DISMISSED** [review]

**Comment by xinyu-intel** [comment]
@tianshuo78520a Hi, we've added test cases for this PR but it seems cannot pass the c++ converage check. Can you give some suggestions on that?

**Comment by xinyu-intel** [comment]
@zoooo0820 @jeff41404 Hi, this is an enhancement for the current fluid operator. Can you please check on it?

File: test/mkldnn/test_fused_vit_attention.py [hunk 1]
```diff
@@ -24,50 +24,53 @@
...
+        token_size = test[1]
+        hidden_size = test[2]
+        num_heads = test[3]
+        dtype = np.float32
 
-        q, k, v = qkv[0], qkv[1], qkv[2]
+        place = paddle.CPUPlace()
+        program = paddle.fluid.Program()
+        startup_program = paddle.fluid.Program()
+        with paddle.fluid.program_guard(program, startup_program):
```
Comment (zoooo0820) [inline test/mkldnn/test_fused_vit_attention.py]: `paddle.static.Program`
`paddle.static.program_guard`

File: test/mkldnn/test_fused_vit_attention.py [hunk 2]
```diff
@@ -24,50 +24,53 @@
...
+            )
 
-    graph = core.Graph(program.desc)
-    core.get_pass("self_attention_fuse_pass").apply(graph)
-    after_program = paddle.fluid.framework.IrGraph(graph).to_program()
-    exe = paddle.static.Executor(place)
-    exe.run(startup_program)
+        graph = core.Graph(program.desc)
+        core.get_pass("self_attention_fuse_pass").apply(graph)
+        after_program = paddle.fluid.framework.IrGraph(graph).to_program()
```
Comment (zoooo0820) [inline test/mkldnn/test_fused_vit_attention.py]: `paddle.framework.IrGraph`

**Review by zoooo0820 — COMMENTED** [review]
Hi， there are some fluid APIs, which will be deprecated very soon. Could you please modify them to our recommended APIs？

By the way, we are planning to remove fluid APIs in this week, and this PRs may be conflicted. Is it possible to merge this PR in next week, thanks.

**Comment by xinyu-intel** [comment]
> Hi， there are some fluid APIs, which will be deprecated very soon. Could you please modify them to our recommended APIs？
> 
> By the way, we are planning to remove fluid APIs in this week, and this PRs may be conflicted. Is it possible to merge this PR in next week, thanks.

Thanks for your suggestions, we'll hold on this PR.

**Comment by zoooo0820** [comment]
Hi, @xinyu-intel @abenmao 
Sorry for blocking this PR before. We have already removed the fluid APIs, now you can just replace these APIs according to the comments, resolve the conflict and continue this PR directly.

**Comment by abenmao** [comment]
> Hi, @xinyu-intel @abenmao Sorry for blocking this PR before. We have already removed the fluid APIs, now you can just replace these APIs according to the comments, resolve the conflict and continue this PR directly.

Done~ But The PR-CI-Api-Benchmark seems to be hung for some reason.

**Review by xinyu-intel — APPROVED** [review]

**Review by YangQun1 — APPROVED** [review]

**Review by vivienfanghuagood — APPROVED** [review]
LGTM

**Pull request merged**